src/components/KTable/KTable.vue

```diff
@@ -845,11 +853,19 @@ export default defineComponent({
     }
 
     watch(() => props.searchInput, (newValue) => {
-      search(newValue)
+      if (newValue === '') {
+        search(newValue)
+      } else {
+        debouncedSearch(newValue)
+      }
```

Comment:
suggestion: Can you add some comments here to explain why we are using two different methods?

Reply:
Comments added